.github/workflows/save-artifacts.yaml

```diff
@@ -18,3 +18,4 @@ jobs:
         with:
           name: artifact
           path: wr_actions
+          retention-days: 1
```

Comment:
Q: why are we saving any artifacts?

Reply:
they're needed for waiting-response label to get added on failed builds

Reply:
Couldn't we do the same thing with an `if: failed()`? Which'd remove the need to store these

Reply:
Using `if: failed()` won't work for forked PRs as they don't have permissions to make the request. We would need to change the the trigger from `pull_request` to `pull_request_target` for it to work, which isn't recommended for security reasons when a checkout is used in the job (see[ here](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/)). It may be ok where we aren't building/running anything but I avoided this in erring on the side of caution.

To get around this we save the pr info (repo name, pr number etc) and instructions to add/remove waiting-response in an artifact. Another workflow is triggered on the completion of the failed workflows which then downloads the artifact and uses the info to add the correct label. I'm open to changing the process as it is a bit convoluted, but I wasn't able to find another way around it without using `pull_request_target` last time I looked into it.

Reply:
Makes sense, this way we can be confident the API requests come from an isolated build